python/salpytools/salpylib.py

```diff
@@ -157,163 +154,220 @@ def reply_to_transition(self, cmdid):
             self.mgr_ackCommand(cmdid, SAL__CMD_COMPLETE, err, message)
 
 
-def validate_transition(current_state, new_state):
-    """
-    Stand-alone function to validate transition. It returns true/false
+class DDSSubscriber(threading.Thread):
+    """Subscribes either to Telemetry or Events and saves the recieved data to 
+    a handed dictionary. In this implementation a new thread is used for every 
+    topic.
+    Attributes:
+        topic: A string of the name of the topic we are subscribing to.
+        handle: An dictionary to save the contents of the recieved data.
+        rate: Rate at which we attempt to obtain new data.
+        timeout: How long to wait for new data before timing out.
     """
-    current_index = csc_states.state_enumeration[current_state]
-    new_index = csc_states.state_enumeration[new_state]
-    transition_is_valid = csc_states.state_matrix[current_index][new_index]
-    if transition_is_valid:
-        LOGGER.info("Transition from {} --> {} is VALID".format(current_state, new_state))
-    else:
-        LOGGER.info("Transition from {} --> {} is INVALID".format(current_state, new_state))
-    return transition_is_valid
+    def __init__(self, topic=None, handle=None, rate=1, timeout=10):
+        threading.Thread.__init__(self)
 
+        self.topic = topic
+        if self.topic is None:
+            raise ValueError("A 'topic' argument must be passed")
 
-class DDSSubscriber(threading.Thread):
+        self.handle = handle
```

Comment:
If the handle can be a simple dictionary, the default behavior could be to create an internal handler if `handle is None` instead of raising an error.

Reply:
If the handle is internal then there would be no way for us to give this data to the caller. Unless we return the dictionary, which then changes the way in which we interact with the library.

Reply:
Not sure I understand what this means. You can aways access the data through `DDSSubscriber.handle`...